be/src/util/string_parser.hpp

```diff
@@ -463,11 +464,23 @@ inline T StringParser::string_to_float_internal(const char* s, int len, ParseRes
                 // overflowing 'remainder' (we stop after 18 instead of 17 to get the rounding
                 // right).
                 if (sig_figs <= 18) {
-                    remainder = remainder * 10 + s[i] - '0';
-                    divide *= 10;
+                    if (LIKELY(acc_num < 63)) {
```

Comment:
The modification will change the logic in cast_expr.cpp. I think we should be more test.